lib/active_admin/filters/resource_extension.rb

```diff
@@ -129,12 +129,36 @@ def default_association_filters
           not_poly.reject! { |r| r.chain.length > 2 }
 
           filters = poly.map(&:foreign_type) + not_poly.map(&:name)
+
+          # Check high-arity associations for filterable columns
+          max = namespace.maximum_association_filter_arity
+          if max != :unlimited
+            high_arity, low_arity = not_poly.partition do |r|
+              r.klass.limit(max + 1).count > max
```

Comment:
Could this be improved by removing the `ORDER BY` clause if present? I think that can be done via `reorder("")`. My concern is that a query with a `LIMIT` can still be slow if it is ordered.

Reply:
Is `r.klass` a scope? I thought it was an AR class (which, I guess, could have a `default_scope`)